### PR TITLE
Adds a check for synced contracts of pfs and client

### LINKED ISSUE #4088: Check PFS infos about registry address
Currently the client doesn't check if the registry address of the PFS (provided in the `info` endpoint) matches the registry the client is running on.

### PR DESCRIPTION
**Pull request by Dominik1999**
closes #4088

### CODE REVIEW

**Comment by Dominik1999** [comment]
@rakanalh I supposed that we want the client to crash when there is something wrong with the chosen PFS. In that case, the client would never receive a route but always pay for the request. Do you agree with the described behaviour here?

File: raiden/tests/integration/network/proxies/test_service_registry.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from raiden.tests.utils.factories import HOP1
 from raiden.tests.utils.smartcontracts import deploy_service_registry_and_set_urls
 from raiden.utils import privatekey_to_address
+from raiden_contracts.constants import CONTRACT_TOKEN_NETWORK_REGISTRY
```
Comment (palango) [inline raiden/tests/integration/network/proxies/test_service_registry.py]: The `CONTRACT_TOKEN_NETWORK_REGISTRY` constant is the name of the contract, not the address.

File: raiden/network/pathfinding.py [hunk 2]
```diff
@@ -179,6 +180,16 @@ def configure_pfs_or_exit(
         )
         click.secho(msg)
         log.info("Using PFS", pfs_info=pathfinding_service_info)
+        pfs_token_network_address = pathfinding_service_info["network_info"]["registry_address"]
+        if pfs_token_network_address != token_network_registry_address:
```
Comment (palango) [inline raiden/network/pathfinding.py]: can you use `is_same_address`  here, it's never clear to me which addresses are encoded in which way. (Something that we need to improve, but not in this PR)

Comment (Dominik1999) [reply raiden/network/pathfinding.py]: good point, changed it

File: raiden/tests/integration/network/proxies/test_service_registry.py (hunk 1 above)
Comment (Dominik1999) [reply raiden/tests/integration/network/proxies/test_service_registry.py]: oh no my bad, sorry changed it to an address

File: raiden/network/pathfinding.py [hunk 3]
```diff
@@ -179,6 +186,16 @@ def configure_pfs_or_exit(
         )
         click.secho(msg)
         log.info("Using PFS", pfs_info=pathfinding_service_info)
+        pfs_token_network_address = pathfinding_service_info["network_info"]["registry_address"]
+        if not is_same_address(pfs_token_network_address, token_network_registry_address):
+            click.secho(
+                f"Invalid reply from pathfinding service {pfs_address}:"
+                f"PFS is not operating on the same Token Network Registry"
+                f"'{pfs_token_network_address}' as you "
```
Comment (rakanalh) [inline raiden/network/pathfinding.py]:
```suggestion
                f"'{pfs_token_network_address}' as your node's "
```

Comment (Dominik1999) [reply raiden/network/pathfinding.py]: thanks for the feedback done

**Pull request merged**